### PR TITLE
[UI] Converted MesheryPerformanceComponent from class-based to functional component

### LINKED ISSUE #8814: [UI] Convert `MesheryPerformanceComponent` from class-based to functional component
### Current Behavior
`MesheryPerformanceComponent` is a class-based component 

### Desired Behavior
`MesheryPerformanceComponent` should be a  functional component.

### Implementation
Convert all the Class-based components to new updated functional components. It is important here to convert a class-based component to a functional component you may need to convert some functions as well. Check if the function is used in other components or not.

Also, make sure to use the hooks while converting them. For example, you need to covert withNotify with useNotification, useRouter hook. Please check if a hook for functional components is available and use that.

Here is a quick blog that will help you get started with : [Link](https://www.digitalocean.com/community/tutorials/five-ways-to-convert-react-class-components-to-functional-components-with-react-hooks)

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by captain-Akshay**
#8975

**Notes for Reviewers**
This PR fixes #8814 

### After the conversion of class based to functional component

[Screencast from 2023-10-01 13-38-49.webm](https://github.com/meshery/meshery/assets/59491379/83598bb7-39f6-462e-a226-01264456f830)

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by captain-Akshay** [comment]
@aabidsofi19  do take a look on this one!

File: ui/components/MesheryPerformance/index.js [hunk 1]
```diff
@@ -712,620 +717,592 @@ class MesheryPerformanceComponent extends React.Component {
...
+                margin="normal"
+                variant="outlined"
+                onChange={handleChange('profileName')}
+                inputProps={{
+                  maxLength: 300,
+                }}
+                InputProps={{
+                  endAdornment: (
+                    <Tooltip title="Create a profile providing a name, if a profile name is not provided, a random one will be generated for you.">
+                      <HelpOutlineOutlinedIcon style={{ color: '#929292' }} />
```
Comment (Yashsharma1911) [inline ui/components/MesheryPerformance/index.js:880]: We should use theme reference color

**Review by Yashsharma1911 — COMMENTED** [review]
Some good work here @captain-Akshay

File: ui/components/MesheryPerformance/index.js (hunk 1 above)
Comment (captain-Akshay) [reply ui/components/MesheryPerformance/index.js:880]: Ok sure!

File: ui/components/MesheryPerformance/index.js [hunk 2]
```diff
@@ -202,108 +225,106 @@ const styles = (theme) => ({
...
+    !(URLValidator(urlState) || isJsonString(additionalOptionsState)),
+  );
+  const [testUUIDState, setTestUUID] = useState(generateUUID());
+  const [selectedMeshState, setSelectedMesh] = useState('');
+  const [availableAdaptersState, setAvailableAdapters] = useState([]);
+  const [availableSMPMeshesState, setAvailableSMPMeshes] = useState([]);
+  // const [disableAvailableOptionsUploadButtonState, setDisableAvailableOptionsUploadButton] =
+  //   useState(false);
+  // const [disableAvailableOptionsInputFieldState, setDisableAvailableOptionsInputField] =
+  //   useState(false);
```
[the remark this answers is not in the document]
Comment (captain-Akshay) [reply ui/components/MesheryPerformance/index.js:292]: OK :+1:

**Comment by captain-Akshay** [comment]
@Yashsharma1911 the two helper functions are only used in this file, & i wasn't able to locate the theme section for tooltip icon color so made it into a class as the other css for consistency

**Review by Ckbhatia — CHANGES_REQUESTED** [review]

File: ui/components/MesheryPerformance/index.js [hunk 3]
```diff
@@ -47,6 +47,7 @@ import { ctxUrl, getK8sClusterIdsFromCtxId } from '../../utils/multi-ctx';
 import { iconMedium } from '../../css/icons.styles';
 import { withNotify } from '../../utils/hooks/useNotification';
 import { EVENT_TYPES } from '../../lib/event-types';
+import { useEffect } from 'react';
```
[the remark this answers is not in the document]
Comment (Ckbhatia) [reply ui/components/MesheryPerformance/index.js:50]: Critical

**Comment by saurabh100ni** [comment]
Thanks for raising the PR, let's Discuss this in the Meshery Development Call today,
Please add this as an agenda in the Doc : [Doc](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit)

**Comment by aabidsofi19** [comment]
@captain-Akshay  is the work here complete

**Comment by aabidsofi19** [comment]
@captain-Akshay  merging this .  please keep an eye on any bugs arising ( incase )

**Pull request merged**